pkg/cgroup/cgroup.go

```diff
@@ -164,23 +171,33 @@ func getOwnCgroupPath(controller string) (string, error) {
 	for s.Scan() {
 		parts := strings.SplitN(s.Text(), ":", 3)
 		if len(parts) < 3 {
-			return "", fmt.Errorf("error parsing /proc/self/cgroup")
+			return nil, fmt.Errorf("error parsing /proc/%s/cgroup", pidStr)
 		}
 		controllerParts := strings.Split(parts[1], ",")
 		for _, c := range controllerParts {
 			if c == controller {
-				return parts[2], nil
+				return parts, nil
 			}
 		}
 	}
 
-	return "", fmt.Errorf("controller %q not found", controller)
+	return nil, fmt.Errorf("controller %q not found", controller)
 }
 
-// createCgroups mounts the cgroup controllers hierarchy for the container but
+// GetOwnCgroupPath returns the cgroup path of this process in controller
+// hierarchy
+func GetOwnCgroupPath(controller string) (string, error) {
+	parts, err := parseCgroupController(-1, controller)
+	if err != nil {
+		return "", err
+	}
+	return parts[2], nil
+}
+
+// CreateCgroups mounts the cgroup controllers hierarchy for the container but
 // leaves the subcgroup for each app read-write so the systemd inside stage1
 // can apply isolators to them
-func createCgroups(root string, subcgroup string, appHashes []types.Hash) error {
```

Comment:
hmm, hang on, this isn't really `pkg` suitable because it has a lot of rkt and stage1 specific implementation details, right? e.g. https://github.com/coreos/rkt/pull/1026/files#diff-df293695bb040b1c55a09afad86cb501R265